bundles/org.openhab.binding.broadlinkthermostat/src/main/java/org/openhab/binding/broadlinkthermostat/internal/handler/FloureonThermostatHandler.java

```diff
@@ -0,0 +1,254 @@
+     */
+    public FloureonThermostatHandler(Thing thing) {
+        super(thing);
+    }
+
+    /**
+     * Initializes a new instance of a {@link FloureonThermostatHandler}.
+     */
+    @Override
+    public void initialize() {
+        super.initialize();
+        try {
+            blDevice = new FloureonDevice(host, new Mac(mac));
+            this.floureonDevice = (FloureonDevice) blDevice;
+        } catch (IOException e) {
+            logger.error("Could not find broadlinkthermostat device at Host {} with MAC {} ", host, mac, e);
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                    "Could not find broadlinkthermostat device at host" + host + "with MAC+" + mac + ": "
+                            + e.getMessage());
+        }
+        authenticate();
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        logger.debug("Command: {}", command.toFullString());
+
+        if (command == RefreshType.REFRESH) {
+            refreshData();
+            return;
+        }
+
+        switch (channelUID.getIdWithoutGroup()) {
+            case SETPOINT:
+                handleSetpointCommand(channelUID, command);
+                break;
+            case POWER:
+                handlePowerCommand(channelUID, command);
+                break;
+            case MODE:
+                handleModeCommand(channelUID, command);
+                break;
+            case SENSOR:
+                handleSensorCommand(channelUID, command);
+                break;
+            case REMOTE_LOCK:
+                handleRemoteLockCommand(channelUID, command);
+                break;
+            case TIME:
+                handleSetTimeCommand(channelUID, command);
+                break;
+            default:
+                logger.warn("Channel {} does not support command {}", channelUID, command);
+        }
+    }
+
+    private void handlePowerCommand(ChannelUID channelUID, Command command) {
+        if (command instanceof OnOffType) {
+            try {
+                floureonDevice.setPower(command == OnOffType.ON);
+            } catch (Exception e) {
+                logger.warn("Error while setting power of {} to {}: {}", thing.getUID(), command, e.getMessage());
+            }
+        } else {
+            logger.warn("Channel {} does not support command {}", channelUID, command);
+        }
+    }
+
+    private void handleModeCommand(ChannelUID channelUID, Command command) {
+        if (command instanceof StringType) {
+            try {
+                if (MODE_AUTO.equals(command.toFullString())) {
+                    floureonDevice.switchToAuto();
+                } else {
+                    floureonDevice.switchToManual();
+                }
+            } catch (Exception e) {
+                logger.error("Error while setting power of {} to {}", thing.getUID(), command, e);
```

Comment:
Logging to error. See above. This could be warn.

Do you log the stack trace by intention?

Please check all.

Reply:
All logs are now on warning and only the exception message is logged.